### PR TITLE
Add info how to install locally

### LINKED ISSUE #2720: Install and use Rollup locally
Documentation Is:

- [x] Missing
- [ ] Needed
- [ ] Confusing
- [ ] Not Sure?

### Please Explain in Detail...

The documentation says [here](https://rollupjs.org/guide/en#creating-your-first-bundle):
> (later on we'll learn how to install it locally to your project so that your build process is portable, but don't worry about that yet)

However, I searched and searched and couldn't find anywhere in the documentation that explains how to do this! 😂

### Your Proposal for Changes

Add a section on how to install and use rollup in a project locally.

### Other

(PS: I'm also having a hard time finding how to do this on Google even! If anyone can help me out with some sweet advice, that would be great!! 🎯)

### PR DESCRIPTION
**Pull request by mesqueeb**
Fixes #2720

This PR contains:
- [ ] bugfix
- [ ] feature
- [ ] refactor
- [x] documentation
- [ ] other

Are tests included?
- [ ] yes (*bugfixes and features will not be merged without tests*)
- [x] no

Breaking Changes?
- [ ] yes (*breaking changes will not be merged unless absolutely necessary*)
- [x] no

List any relevant issue numbers:

### Description

Fixes #2720

### CODE REVIEW

File: docs/04-tutorial.md [hunk 1]
```diff
@@ -127,6 +127,33 @@ rollup --config rollup.config.dev.js
...
+```console
+./node_modules/.bin/rollup --config
+```
+
+In this case it's best to just add this command as a script to your `package.json`:
+
+```json
+{
+  "scripts": {
+    "build": "./node_modules/.bin/rollup --config"
```
Comment (TrySound) [inline docs/04-tutorial.md]: This is not right. npm and yarn will resolve bin for you. Direct path will stop work with yarn v2.

Comment (mesqueeb) [reply docs/04-tutorial.md]: @TrySound Can you please update the PR with the correct information? I really want to know how to use Rollup locally both with NPM and Yarn.

Currently what I wrote I got from [this post](https://github.com/rollup/rollup/issues/2720#issuecomment-467357816) and they asked me to make the PR on the docs.

Currently there's no mention in the docs anywhere how to use a local rollup installation.

Comment (TrySound) [reply docs/04-tutorial.md]: The way is identical for any package with cli
```
yarn add package
yarn package
npm install package
npm run package
```

https://yarnpkg.com/en/docs/cli/#locally-installed-clis

**Comment by mesqueeb** [comment]
@TrySound 
I have updated the branch to reflect your feedback!
Kindly review and advise. ;)

File: docs/04-tutorial.md [hunk 2]
```diff
@@ -141,15 +141,16 @@ npm i -D rollup
 Now you can run it inside the root folder of your project like so:
 
 ```console
-./node_modules/.bin/rollup --config
+npm run rollup --config
```
Comment (lukastaegert) [inline docs/04-tutorial.md]: This is not how it works for npm, I assume this is a misunderstanding (and I would seriously recommend to try it out yourself on your machine). `npm run` (which is short form of `npm run-script`) will only run scripts defined in your package.json file but not files in your `.bin` folder, see https://docs.npmjs.com/cli/run-script.html

To run files from `node_modules/bin`, `npx` is as far as I know the best choice. What many find confusing is that if you forgot to install the file, then `npx` will download it for you. But a local file will always have precedence. For some reason, npm fails to properly document this on their home page but there is a nice intro in their blog: https://blog.npmjs.org/post/162869356040/introducing-npx-an-npm-package-runner

File: docs/04-tutorial.md [hunk 3]
```diff
@@ -141,15 +141,16 @@ npm i -D rollup
...
+// OR yarn rollup --config
```
Comment (lukastaegert) [inline docs/04-tutorial.md]: This, however, seems to be correct as yarn does not seem to discern as stubbornly between scripts and executables.

Comment (TrySound) [reply docs/04-tutorial.md]: Yeah, this just works with yarn

File: docs/04-tutorial.md (hunk 2 above)
Comment (tivac) [reply docs/04-tutorial.md]: `npx rollup --config` will work though.

**Comment by mesqueeb** [comment]
@lukastaegert @tivac @TrySound 
I have updated the PR with `npx` instead of `npm run`

File: docs/04-tutorial.md [hunk 4]
```diff
@@ -127,6 +127,34 @@ rollup --config rollup.config.dev.js
...
+npx rollup --config
+// OR yarn rollup --config
+```
+
+In this case it's usually a good idea to also add a script to your `package.json`, so the whole team uses the correct build command:
+
+```json
+{
+  "scripts": {
+    "build": "npx rollup --config"
```
Comment (tivac) [inline docs/04-tutorial.md]: This version doesn't need to use `npx`, it can simply be `"build": "rollup --config"`

**Pull request merged**